machine_learning/loss_functions/categorical_cross_entropy.py

```diff
@@ -0,0 +1,58 @@
+"""
+Categorical Cross-Entropy Loss
+
+This function calculates the Categorical Cross-Entropy Loss between true class
+labels and predicted class probabilities.
+
+Formula:
+Categorical Cross-Entropy Loss = -Σ(y_true * log(y_pred))
+
+Resources:
+- [Wikipedia - Cross entropy](https://en.wikipedia.org/wiki/Cross_entropy)
+"""
+
+import numpy as np
+
+
+def categorical_crossentropy(
+    y_true: np.ndarray, y_pred: np.ndarray, epsilon: float = 1e-15
+) -> float:
+    """
+    Calculate Categorical Cross-Entropy Loss between true class labels and
+    predicted class probabilities.
+
+    Parameters:
+    - y_true: True class labels (one-hot encoded) as a NumPy array.
```

Comment:
Could we have a check to enforce that `y_true` is actually one-hot encoded? For example, the function should see an input like `[2, 0, 1]` and raise a ValueError.